src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java

```diff
@@ -73,7 +74,17 @@ public InventoryMock(@Nullable InventoryHolder holder, @NotNull InventoryType ty
 		this.holder = holder;
 		this.type = type;
 
-		items = new ItemStack[type.getDefaultSize()];
+		items = new ItemStack[inventoryType2StorageContentsSize(type)];
+	}
+
+	private static int inventoryType2StorageContentsSize(InventoryType type)
+	{
+		return switch (type)
+		{
+			// workbench has 1 result slot
+			case WORKBENCH -> type.getDefaultSize() - 1;
+			default -> type.getDefaultSize();
+		};
```

Comment:
I had to change this because using `type.getDefaultSize()` would create arrays larger than necessary, as result slots are not in this array. This also caused a unit test to fail in `InventoryMockTest` because there `getSize()` is not implemented adequately.